src/modules/entities/entities-saga.ts

```diff
@@ -0,0 +1,14 @@
+import {User} from "../users/user-types";
+import {normalize} from "normalizr";
+import {user} from "./entities-schema";
+import {put} from "redux-saga/effects";
+import {entitiesActions} from "./entities-slice";
+import {NormUser, UserEntities} from "./normalized-user-types";
+
+export function* normalizeAndStore(data: User[]) {
+    const normData = normalize<NormUser, UserEntities, string[]>(data, [user]);
+
+    yield put(entitiesActions.entitiesUpdated(normData.entities));
+
+    return normData.result;
+}
```

Comment:
Toto bych udělal trochu obecněji. Bylo by fajn, kdyby se schéma dalo definovat při volání této ságy, aby ta sága byla víc univerzální a dala se použít na víc věcí v appce.

```suggestion
export function* normalizeAndStore(data: unknown, schema: Schema) {
    const normData = normalize<unknown, UserEntities>(data, schema);

    yield put(entitiesActions.entitiesUpdated(normData.entities));

    return normData.result;
}
```

Reply:
Místo `UserEntities` by se později dal použít ještě nějaký jiný interface, který by slučoval všechny typy entit v celé appce, ale my tam máme jen `UserEntities`.